clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -133,6 +133,7 @@ public boolean hasNotJoinedGroup() {
     protected final ConsumerNetworkClient client;
 
     private Node coordinator = null;
+    private String rejoinReason = "initialized abstract coordinator";
```

Comment:
The reason is a bit weird. I wonder if we could just leave it empty in the beginning.